yt_idefix/data_structures.py

```diff
@@ -280,6 +281,10 @@ def _parse_inifile(self) -> None:
         if self.inifile is None:
             return
 
+        if "Grid" in self.parameters:
+            # In case inifile was already parsed
+            return
+
```

Comment:
If we end up parsing multiple times then the problem is elsewhere, I don't think we should guard against it here.

Reply:
The comment I left here might be not exact. 

Since I call `self._parse_inifile()` in advance (but not always) in `self._parse_header_file()` which is called in `self.__init__()`, after which, however, the original `self._parse_inifile()` is called. Therefore here I avoid the inifile is parsed again.

If we could just only parse the inifile in `self.__init__()`, the problem will not exist.

Reply:
Then let's do that

Reply:
I found the best place to parse the inifile once is after `self.parameters` being created and before definitions.h being read, which should be somewhere in `Dataset.__init__()`. So I think we can't avoid this part.